tests/check_framework/test_translation.py

```diff
@@ -33,9 +34,66 @@ def test_invalid_language_code(self):
             # FIXME: The following should be invalid:
             # 'sr@latin',      # locale instead of language tag.
         )
-        for tag in tags:
+
+    def test_valid_language_code(self):
+        for tag in self.valid_tags:
+            with self.subTest(tag), override_settings(LANGUAGE_CODE=tag):
+                self.assertEqual(check_setting_language_code(None), [])
+
+    def test_invalid_language_code(self):
+        msg = 'You have provided an invalid value for the LANGUAGE_CODE setting: %s.'
+        for tag in self.invalid_tags:
             with self.subTest(tag), override_settings(LANGUAGE_CODE=tag):
                 result = check_setting_language_code(None)
-                self.assertEqual(result, [E001])
+                error = copy(E001)
+                error.msg = error.msg.format(tag)
+                self.assertEqual(result, [error])
                 self.assertEqual(result[0].id, 'translation.E001')
-                self.assertEqual(result[0].msg, 'You have provided an invalid value for the LANGUAGE_CODE setting.')
+                self.assertEqual(result[0].msg, msg % tag)
```

Comment:
I think these two checks:
```python
self.assertEqual(result[0].id, 'translation.E001') 
self.assertEqual(result[0].msg, msg % tag) 
```
are redundant with
```python
self.assertEqual(result, [error])
```
(in all tests). I think we can add explicit check (like in other check framework's tests) e.g.
```python
self.assertEqual(check_setting_languages(None), [
    Error(
        'You have provided an invalid value for the '
        'LANGUAGE_CODE setting: %s.' % tag,
        id='translation.E001',
    ),
])
```

Reply:
Yeah, I think this pattern came from one of the other check framework modules. It would probably make sense to change these.